client/src/pages/user/Profile.js

```diff
@@ -16,16 +14,28 @@ const Profile = () => {
 
   //get user data
   useEffect(() => {
-    const { email, name, phone, address } = auth?.user;
-    setName(name);
-    setPhone(phone);
-    setEmail(email);
-    setAddress(address);
+    if (!auth || !auth.user) {
+      toast.error("Auth Token is not found. Please sign out and sign in again.");
+      return
+    }
+    else {
+      const { email, name, phone, address, password } = auth?.user;
+      setName(name);
+      setPhone(phone);
+      setEmail(email);
+      setPassword(password);
+      setAddress(address);
+    }
   }, [auth?.user]);
```

Comment:
I think this should just be 
```suggestion
  }, [auth]);
```